core/src/main/scala/kafka/server/DelayedElectPreferredLeader.scala

```diff
@@ -55,10 +55,6 @@ class DelayedElectPreferredLeader(delayMs: Long,
     responseCallback(timedout ++ fullResults)
   }
 
-  private def timeoutWaiting = {
-    waitingPartitions.map(partition => partition -> new ApiError(Errors.REQUEST_TIMED_OUT, null)).toMap
-  }
-
```

Comment:
@tombentley Please check that we didn't intend to use this.

Reply:
Yes, this can be removed since the logic is folded into onComplete() already.